go-controller/pkg/ovn/policy.go

```diff
@@ -148,13 +171,40 @@ func (oc *Controller) updateStaleDefaultDenyACLNames(npType knet.PolicyType, gre
 }
 
 func (oc *Controller) syncNetworkPolicies(networkPolicies []interface{}) error {
+	// first see if this is post "support shared port group" by trying to find there
+	// is any default deny port group created the expected external_ids
+	pgp := func(item *nbdb.PortGroup) bool {
+		return item.ExternalIDs["support_shared_port_group"] == "yes"
+	}
+	spgDefaultDenyPGs, err := libovsdbops.FindPortGroupsWithPredicate(oc.nbClient, pgp)
+	if err != nil {
+		return fmt.Errorf("failed to get default deny port groups that created with shared port group support: %v", err)
+	}
+	defaultDenyPGMap := map[string]bool{}
+	for _, pg := range spgDefaultDenyPGs {
+		defaultDenyPGMap[pg.Name] = true
+	}
+	stalePGs := sets.NewString()
 	expectedPolicies := make(map[string]map[string]bool)
 	for _, npInterface := range networkPolicies {
 		policy, ok := npInterface.(*knet.NetworkPolicy)
 		if !ok {
 			return fmt.Errorf("spurious object in syncNetworkPolicies: %v", npInterface)
 		}
 
+		// If the current port groups were created before shared group support, we need to remove
```

Comment:
I guess this could create a temporary security hole while we sync then wait to add the policies back, but not sure what else we could do without getting more complicated